### PR TITLE
Display message fix

### PR DESCRIPTION
**Pull request by VALERA771**
Fix for #1748. Tested

### CODE REVIEW

File: Exiled.CustomItems/API/Features/CustomItem.cs [hunk 1]
```diff
@@ -774,8 +775,7 @@ public virtual void Give(Player player, Item item, bool displayMessage = true)
                 if (!TrackedSerials.Contains(item.Serial))
                     TrackedSerials.Add(item.Serial);
 
-                if (displayMessage)
-                    ShowPickedUpMessage(player);
+                this.displayMessage = displayMessage;
```
Comment (Misaka-ZeroTwo) [inline Exiled.CustomItems/API/Features/CustomItem.cs:778]:
```suggestion
```

File: Exiled.CustomItems/API/Features/CustomItem.cs [hunk 2]
```diff
@@ -56,6 +56,7 @@ public abstract class CustomItem
         private static Dictionary<Type, CustomItem?> typeLookupTable = new();
         private static Dictionary<string, CustomItem?> stringLookupTable = new();
         private static Dictionary<uint, CustomItem?> idLookupTable = new();
+        private bool displayMessage;
```
Comment (Misaka-ZeroTwo) [inline Exiled.CustomItems/API/Features/CustomItem.cs:59]:
```suggestion
```

File: Exiled.CustomItems/API/Features/CustomItem.cs [hunk 3]
```diff
@@ -999,7 +999,11 @@ protected virtual void OnUpgrading(API.EventArgs.UpgradingItemEventArgs ev)
         /// Called anytime the item enters a player's inventory by any means.
         /// </summary>
         /// <param name="player">The <see cref="Player"/> acquiring the item.</param>
-        protected virtual void OnAcquired(Player player) => ShowPickedUpMessage(player);
+        protected virtual void OnAcquired(Player player) // TODO: Male field 'displayMessage' parameter
+        {
+            if (displayMessage)
+                ShowPickedUpMessage(player);
+        }
```
Comment (Misaka-ZeroTwo) [inline Exiled.CustomItems/API/Features/CustomItem.cs:1006]:
```suggestion
        protected virtual void OnAcquired(Player player) => ShowPickedUpMessage(player);
```

File: Exiled.CustomItems/API/Features/CustomItem.cs [hunk 4]
```diff
@@ -774,8 +775,7 @@ public virtual void Give(Player player, Item item, bool displayMessage = true)
                 if (!TrackedSerials.Contains(item.Serial))
                     TrackedSerials.Add(item.Serial);
 
-                if (displayMessage)
-                    ShowPickedUpMessage(player);
+                this.displayMessage = displayMessage;
 
                 Timing.CallDelayed(0.05f, () => OnAcquired(player));
```
Comment (Misaka-ZeroTwo) [inline Exiled.CustomItems/API/Features/CustomItem.cs:780]:
```suggestion
```

**Review by Misaka-ZeroTwo — COMMENTED** [review]
There's no need to call OnAcquired at this point.

**Comment by VALERA771** [comment]
> There's no need to call OnAcquired at this point.

So then if item will be added with method users users won't be able to do their stuff

File: Exiled.CustomItems/API/Features/CustomItem.cs [hunk 5]
```diff
@@ -999,7 +999,11 @@ protected virtual void OnUpgrading(API.EventArgs.UpgradingItemEventArgs ev)
         /// Called anytime the item enters a player's inventory by any means.
         /// </summary>
         /// <param name="player">The <see cref="Player"/> acquiring the item.</param>
-        protected virtual void OnAcquired(Player player) => ShowPickedUpMessage(player);
+        protected virtual void OnAcquired(Player player) // TODO: Male field 'displayMessage' parameter
```
Comment (Misaka-ZeroTwo) [inline Exiled.CustomItems/API/Features/CustomItem.cs:1002]: What does this TODO means?

**Comment by Misaka-ZeroTwo** [comment]
> > There's no need to call OnAcquired at this point.
> 
> So then if item will be added with method users users won't be able to do their stuff

oopsie, my bad.

**Review by Misaka-ZeroTwo — APPROVED** [review]

File: Exiled.CustomItems/API/Features/CustomItem.cs (hunk 5 above)
Comment (VALERA771) [reply Exiled.CustomItems/API/Features/CustomItem.cs:1002]: If add a parameter to method all overridden method will stop working

Comment (Misaka-ZeroTwo) [reply Exiled.CustomItems/API/Features/CustomItem.cs:1002]: Yeah I see, just confused about you used the word Male

Comment (xNexus-ACS) [reply Exiled.CustomItems/API/Features/CustomItem.cs:1002]: It means to make the displayMessage field into a parameter of the method

Comment (VALERA771) [reply Exiled.CustomItems/API/Features/CustomItem.cs:1002]: > Yeah I see, just confused about you used the word Male
> 
> 

Sry for my English ;)

Comment (Misaka-ZeroTwo) [inline Exiled.CustomItems/API/Features/CustomItem.cs]:
```suggestion
        protected virtual void OnAcquired(Player player) // TODO: Add 'displayMessage' parameter and remove the field in the class
```

**Pull request merged**